pmm/pmm2-release.groovy

```diff
@@ -36,8 +36,149 @@ pipeline {
+                                        createrepo --update /srv/\${REPOPATH}/release/\${rhel}/SRPMS
+                                        if [ -f /srv/\${REPOPATH}/release/\${rhel}/SRPMS/repodata/repomd.xml.asc ]; then
+                                            rm -f /srv/\${REPOPATH}/release/\${rhel}/SRPMS/repodata/repomd.xml.asc
+                                        fi
+                                        gpg --detach-sign --armor --passphrase ${SIGN_PASSWORD} /srv/\${REPOPATH}/release/\${rhel}/SRPMS/repodata/repomd.xml
+                                    done
+                                    fi
+
+                                    # binary processing
+                                    pushd binary
+                                    for rhel in \${RHVERS}; do
+                                        mkdir -p /srv/\${REPOPATH}/release/\${rhel}/RPMS
+                                        for arch in $(ls -1 redhat/\${rhel}); do
+                                            mkdir -p /srv/\${REPOPATH}/release/\${rhel}/RPMS/\${arch}
+                                            cp -av redhat/\${rhel}/\${arch}/*.rpm /srv/\${REPOPATH}/release/\${rhel}/RPMS/\${arch}/
+                                            createrepo --update /srv/\${REPOPATH}/release/\${rhel}/RPMS/\${arch}/
+                                            if [ -f  /srv/\${REPOPATH}/release/\${rhel}/RPMS/\${arch}/repodata/repomd.xml.asc ]; then
+                                                rm -f  /srv/\${REPOPATH}/release/\${rhel}/RPMS/\${arch}/repodata/repomd.xml.asc
+                                            fi
+                                            gpg --detach-sign --armor --passphrase $SIGN_PASSWORD /srv/\${REPOPATH}/release/\${rhel}/RPMS/\${arch}/repodata/repomd.xml
+                                        done
+                                    done
+                                done
+                            ENDSSH
+                        """
+                    }
+                }
+            }
+        }
+        stage('Push DEB client to public repository') {
+            steps {
+                withCredentials([string(credentialsId: 'SIGN_PASSWORD', variable: 'SIGN_PASSWORD')]) {
+                    withCredentials([sshUserPrivateKey(credentialsId: 'repo.ci.percona.com', keyFileVariable: 'KEY_PATH', passphraseVariable: '', usernameVariable: 'USER')]) {
+                        sh """
+                            ssh -o StrictHostKeyChecking=no -i ${KEY_PATH} ${USER}@repo.ci.percona.com << 'ENDSSH'
+                                set -e
+                                #
+                                #
+                                REPOS='PERCONA TOOLS PMM2-CLIENT'
+                                for REPOSITORY in \$REPOS; do
+                                    if [[ \${REPOSITORY} = PERCONA ]]; then
+                                        REPOPATH=/srv/repo-copy/percona/apt
```

Comment:
I think we need also to set default PATH here as there might be an issue if PATH already includes reprepro5

Reply:
I added default: `export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin`

Reply:
here is the default PATH on the server
uploader@repo:~$ echo $PATH
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/opt/puppetlabs/bin